core/Services/Calendar/Store.vala

```diff
@@ -59,22 +61,12 @@ public class Calendar.Store : Object {
     public GLib.Queue<E.Source> calendar_trash;
     private E.CredentialsPrompter credentials_prompter;
 
-    private static Calendar.Store? store = null;
-    private static GLib.Settings? state_settings = null;
-
-    public static Calendar.Store get_default () {
-        if (store == null)
-            store = new Calendar.Store ();
-        return store;
-    }
+    protected static Calendar.Store? store = null;
+    protected static GLib.Settings? state_settings = null;
 
-    static construct {
-        if (SettingsSchemaSource.get_default ().lookup ("io.elementary.calendar.savedstate", true) != null) {
-            state_settings = new GLib.Settings ("io.elementary.calendar.savedstate");
-        }
-    }
+    protected Store (string source_extension_name) {
+        Object (source_extension_name: source_extension_name);
 
```

Comment:
This seems to be the source of your issue with the test. I don't think anything after the `Object (...)` block is actually being run ([docs](https://wiki.gnome.org/Projects/Vala/Tutorial#GObject-Style_Construction)). Here's what fixed it for me:

```suggestion
    protected Store (string source_extension_name) {
        Object (source_extension_name: source_extension_name);
    }
    
    construct {
```

Reply:
@mcclurgm good catch!! Coming from Java, I wasn't aware of this behaviour 🎉️